internal/controller/volsync/vshandler.go

```diff
@@ -1027,6 +1027,13 @@ func (v *VSHandler) EnsurePVCforDirectCopy(ctx context.Context,
 		},
 	}
 
+	volumeMode := corev1.PersistentVolumeFilesystem
+	if util.IsRBDEnabledForVolSyncReplication(v.owner.GetAnnotations()) &&
+		rdSpec.ProtectedPVC.VolumeMode != nil &&
+		*rdSpec.ProtectedPVC.VolumeMode == corev1.PersistentVolumeBlock {
+		volumeMode = corev1.PersistentVolumeBlock
+	}
```

Comment:
This need to be a helper - the details of how we determine the volume mode are not interesting in this context, and make following the code harder.

We already disabled the linter here, we should not make this function worse.

Reply:
Point taken.

Reply:
You are correct. I'll fix it.